docs/deployment/teleport.md

```diff
@@ -0,0 +1,24 @@
+# Authentication with Teleport
+
+Setting up Teleport authentication will protect all routes of your notebook. It is particularly useful for adding authentication to deployed notebooks. Teleport authentication is provided in addition to [Livebook's authentication](../authentication.md) for authoring notebooks.
+
+## How to
+
+To integrate Teleport authentication with Livebook,
+set the `LIVEBOOK_IDENTITY_PROVIDER` environment variable to `LIVEBOOK_IDENTITY_PROVIDER=teleport:https://teleport-cluster.example.com/.well-known/jwks.json`, make sure the `teleport` CLI is installed and available on your machine (or your Docker image).
```

Comment:
I assume you are not supposed to use example.com, right? Shall we add a line that asks to replace `.example.com` by something else?

Reply:
Why do we need the `teleport` CLI? I don't see us using it in the code.

Reply:
> I assume you are not supposed to use example.com, right?

Correct, each organizations have their own teleport setup, so I'll replace it with something like `https://your-cluster.hostname.com/.well-known/jwks.json` ?

BTW I'm not sure if I should just ask for `https://your-cluster.hostname.com` and assume the path is `/.well-known/jwks.json`? 

> Why do we need the teleport CLI? I don't see us using it in the code.

Right, I copied this doc from Tailscale docs 😬 

But at the same time you need to have [teleport running](https://goteleport.com/docs/application-access/guides/connecting-apps/#start-the-application-service-with-cli-flags) somewhere installed, in my case I used it on my laptop. It's the same for Tailscale so I thought I should put it here!

Reply:
> BTW I'm not sure if I should just ask for https://your-cluster.hostname.com and assume the path is /.well-known/jwks.json?

Yeah, let's do that for now.